### PR TITLE
People page

### PR DESCRIPTION
**Pull request by mileni67**
I need help, I  thought that "pagination" would be done differently, and now i don't know koh to do it right. ;( Can you help me? My brain is fried already(

### CODE REVIEW

**Comment by mileni67** [comment]
i imagined that pagination would be implemented using "slice" and there would ne something that: 

reducers: {
    changePage: (state, { payload: action }) => {
      state.page = action;
    },
    resetPage: (state) => {
      state.page = 1;
    },
  },
});
export const { changePage, resetPage } = paginationSlice.actions;

export const selectPaginationState = (state) => state.pagination;
export const selectPage = (state) => selectPaginationState(state).page;

**Comment by Siedemus** [comment]
i imagined that you would create a slice similar to movies slice:

`import { createSlice } from "@reduxjs/toolkit";

const initialState = {
  status: "loading",
  data: {},
  currentPage: 1,
};

const moviesSlice = createSlice({
  name: "movies",
  initialState,
  reducers: {
    fetchMovies: (state) => {
      state.status = "loading";
    },
    setMovies: (state, { payload }) => {
      state.data = payload;
      state.status = "success";
    },
    setError: (state) => {
      state.status = "error";
    },
    firstMoviePage: (state) => {
      state.currentPage = 1;
    },
    previousMoviePage: (state) => {
      if (state.currentPage !== 1) {
        state.currentPage--;
      }
    },
    nextMoviePage: (state) => {
      if (state.currentPage !== 500) {
        state.currentPage++;
      }
    },
    lastMoviePage: (state) => {
      state.currentPage = 500;
    },
    setMoviePageByQuery: (state, { payload }) => {
      if (isNaN(payload)) {
        state.currentPage = 1;
      } else if (payload > 500) {
        state.currentPage = 500;
      } else if (payload < 1) {
        state.currentPage = 1;
      } else {
        state.currentPage = payload;
      }
    },
  },
});

export const {
  fetchMovies,
  setMovies,
  setError,
  firstMoviePage,
  previousMoviePage,
  nextMoviePage,
  lastMoviePage,
  setMoviePageByQuery,
} = moviesSlice.actions;

const selectMoviesState = (state) => state.movies.data;

export const selectMoviesList = (state) => selectMoviesState(state).results;

export const selectCurrentMoviePage = (state) => state.movies.currentPage;

export default moviesSlice.reducer;
`

and next we just need to push people page reducers to pagination as props like here:

`
  return (
    <>
      <Container>
        <Header>Popular movies</Header>
        <MovieList>
          {moviesData
            ? moviesData.map((movie) => (
                <Movie key={movie.poster_path}>
                  <MainTile
                    id={movie.id}
                    poster={movie.poster_path}
                    title={movie.title}
                    subtitle={movie.release_date.slice(0, 4)}
                    tags={movie.genre_ids}
                    rate={{
                      score: movie.vote_average,
                      votes: movie.vote_count,
                    }}
                  />
                </Movie>
              ))
            : null}
        </MovieList>
        <Pagination
          currentPage={currentMoviePage}
          firstPage={firstMoviePage}
          previousPage={previousMoviePage}
          nextPage={nextMoviePage}
          lastPage={lastMoviePage}
        />
      </Container>
    </>
  );
};
`

**Comment by Siedemus** [comment]
You just need to create currentPage in initalState in peopleSlice and set it to 1, and you also need to create 4 reducers: firstPeoplePage, previousPeoplePage, nextMoviePage, lastMoviePage. Next, you need to add pagination to peoplePage and push these five things to this component, and all will work.

File: src/Features/Navigation/Search/index.js [hunk 1]
```diff
@@ -1,11 +1,39 @@
...
+import { useQueryParameter, useReplaceQueryParameter } from "./queryParameter";
+/*import { firstPage } from "Pagination";*/
 
 export const Search = () => {
+    const dispatch = useDispatch();
+    const moviesMatch = useRouteMatch("/movies");
+    const peopleMatch = useRouteMatch("/people");
+    const placeholder = moviesMatch
+        ? "Search for movies..."
+        : "Search for people...";
```
Comment (DominikaBuczekSzkowron) [inline src/Features/Navigation/Search/index.js:15]: good point 👍

File: src/Features/PeoplePage/getDataApi.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
+import axios from "axios";
+
+const URL = "https://api.themoviedb.org/3";
+const API_KEY = "5d7b278fff9e1056c09e514d9f0bc9f3";
```
Comment (DominikaBuczekSzkowron) [inline src/Features/PeoplePage/getDataApi.js:4]: I think that woud be better to use file apiURLs.js

File: src/App.js [hunk 3]
```diff
@@ -5,14 +5,14 @@ import {
   Redirect,
 } from "react-router-dom/cjs/react-router-dom";
 import MoviesPage from "./Features/MoviesPage";
-import PeoplePage from "./Features/PeoplePage";
+import { PeoplePage } from "./Features/PeoplePage";
```
Comment (DominikaBuczekSzkowron) [inline src/App.js:8]: the name of this component is PeopleList. Try:
import { PeopleList } from "./Features/PeoplePage";

File: src/Features/PeoplePage/index.js [hunk 4]
```diff
@@ -1,3 +1,60 @@
...
+import { PeopleListTile } from "../../common/TileList/TilePeople/index";
+import { Container } from "../../common/Container";
+import {
+    selectPeopleList,
+    selectTotalResult,
+    fetchSearchPeopleLoad,
+    fetchPeopleListLoad,
+} from "./peopleSlice";
+import { searchQueryParamName } from "../Navigation/Search/searchQueryParamName";
+import { Pagination } from "../../common/Pagination";
```
Comment (DominikaBuczekSzkowron) [inline src/Features/PeoplePage/index.js:15]: I thought that Bartek is working on pagination.

**Review by DominikaBuczekSzkowron — CHANGES_REQUESTED** [review]
It seems that you put a lot of effort to your work. I think it's hard to do every task in one time. It's much easier is shearing all work to small steps. It's easier and it's much better, e.g. first you try to run PeoplePage, push the branch even if it's not running, when the app is running do pull request and than merge that branch with main. After that your teammates can switch their work to yours. e.g. Bartek could add pagination to your PeoplePage. I could work on PeopleDetails, loading or sth else. The app in this way is building and there is better effect of the result.  Product Owner can comment our work and there is much more time to correct of bugs. Just after merging PeoplePage you can start work on Searching.
I can't run your changes. I see you tried do everything on one branch: PeoplePage, Searching even Pagination. I think the method step by step is the best in IT.

**Pull request closed without merging**